### PR TITLE
Fix elastic-agent-service Docker container entrypoint

### PR DESCRIPTION
**Pull request by cmacknz**
The `elastic-agent-service` container followed the template for the `elastic-agent-cloud` container. This isn't quite correct, because while the service container will run exclusively in cloud it is otherwise meant to be used as a standard Elastic Agent docker container with the standard entrypoint.

- Closes https://github.com/elastic/ingest-dev/issues/4363#issuecomment-2447236159

### CODE REVIEW

**Comment by cmacknz** [comment]
Added both original reviewers from https://github.com/elastic/elastic-agent/pull/5349 which introduced this image since Blake is currently out.

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 1]
```diff
@@ -42,7 +42,7 @@ RUN true && \
     chmod 0775 {{ $beatHome}}/{{ $modulesd }} && \
 {{- end }}
 
-{{- if or (eq .Variant "cloud") (eq .Variant "service") }}
+{{- if or (eq .Variant "cloud") }}
```
Comment (pkoutsovasilis) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:45]: no need for the `or` any more right?

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 2]
```diff
@@ -172,7 +172,7 @@ RUN mkdir /licenses
 COPY --from=home {{ $beatHome }}/LICENSE.txt /licenses
 COPY --from=home {{ $beatHome }}/NOTICE.txt /licenses
 
-{{- if or (eq .Variant "cloud") (eq .Variant "service") }}
+{{- if or (eq .Variant "cloud") }}
```
Comment (pkoutsovasilis) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:175]: you know 🙂 (above)

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 3]
```diff
@@ -294,7 +294,7 @@ ENV LIBBEAT_MONITORING_CGROUPS_HIERARCHY_OVERRIDE=/
 
 WORKDIR {{ $beatHome }}
 
-{{- if or (eq .Variant "cloud") (eq .Variant "service") }}
+{{- if or (eq .Variant "cloud") }}
```
Comment (pkoutsovasilis) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:297]: you know 🙂 (above)

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 1 above)
Comment (cmacknz) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:45]: How did I miss this, thanks!

**Review by pkoutsovasilis — APPROVED** [review]
in the original PR I did review the changes that affected the existing k8s integration tests. Full disclosure, I am not entirely sure why the the `command` and `args` had to change for the `TestKubernetesAgentService` but from what I can tell there are not needed. Thus LGTM 🙂

**Review by pchila — APPROVED** [review]
LGTM

**Pull request merged**